src/openrct2/ride/Vehicle.cpp

```diff
@@ -738,6 +740,34 @@ template<> bool SpriteBase::Is<Vehicle>() const
     return Type == EntityType::Vehicle;
 }
 
+/**
+ * Fires the "vehicle.crash" api hook
+ * @param vehicleId Entity id of the vehicle that just crashed
+ * @param crashId What the vehicle crashed into. Should be either "another_vehicle", "land", or "water"
+ */
+static void InvokeVehicleCrashHook(const uint16_t vehicleId, const std::string_view crashId)
+{
+    // If ENABLE_SCRIPTING is not defined, the body of this function is empty.
+    // Calls to this function should probably still be wrapped in a #ifdef to avoid
+    //  unoptimized code from wasting time by calling an empty function
+#ifdef ENABLE_SCRIPTING
```

Comment:
You need to wrap the entire function or you'll get warnings about unused functions when scripting is disabled.